source/extensions/filters/common/expr/context.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "envoy/stream_info/stream_info.h"
+
+#include "common/http/headers.h"
+
+#include "eval/public/cel_value.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+
+using CelValue = google::api::expr::runtime::CelValue;
+
+// Symbols for traversing the request properties in the expressions
+constexpr absl::string_view Request = "request";
+constexpr absl::string_view Path = "path";
+constexpr absl::string_view Host = "host";
+constexpr absl::string_view Scheme = "scheme";
+constexpr absl::string_view Method = "method";
+constexpr absl::string_view Referer = "referer";
+constexpr absl::string_view Headers = "headers";
+constexpr absl::string_view Time = "time";
+constexpr absl::string_view ID = "id";
+constexpr absl::string_view UserAgent = "useragent";
+constexpr absl::string_view Size = "size";
+constexpr absl::string_view TotalSize = "total_size";
+
+// Per-request metadata
+constexpr absl::string_view Metadata = "metadata";
+
+// Downstream connection properties
+constexpr absl::string_view Connection = "connection";
+constexpr absl::string_view LocalAddress = "local_address";
+constexpr absl::string_view RemoteAddress = "remote_address";
+
+class RequestWrapper;
+
+class HeadersWrapper : public google::api::expr::runtime::CelMap {
```

Comment:
Why doesn't `HeadersWrapper` subclass `BaseWrapper`?

Reply:
It provides the meaning for size and empty.